relayer/relays/beacon/header/header.go

```diff
@@ -244,127 +202,75 @@ func (h *Header) SyncHeader(ctx context.Context, headerUpdate scale.HeaderUpdate
 	if err != nil {
 		return err
 	}
 
 	if !hasChanged {
 		return ErrFinalizedHeaderUnchanged
 	}
-
-	sync, err := h.SyncFinalizedHeader(ctx)
-	if err != nil {
-		return err
+	fromSlot := h.cache.Finalized.LastSyncedExecutionSlot
+	if fromSlot == 0 {
+		fromSlot = h.cache.Finalized.LastSyncedSlot
 	}
-	err = h.SyncHeaders(ctx, secondLastFinalizedHeader, sync.FinalizedHeaderBlockRoot)
 
+	err = h.SyncFinalizedHeader(ctx)
 	if err != nil {
 		return err
 	}
 
-	return nil
-}
+	err = h.SyncHeaders(ctx, fromSlot, h.cache.Finalized.LastSyncedSlot)
 
-func (h *Header) SyncHeaders(ctx context.Context, fromHeaderBlockRoot, toHeaderBlockRoot common.Hash) error {
-	fromHeader, err := h.syncer.Client.GetHeader(fromHeaderBlockRoot)
 	if err != nil {
 		return err
 	}
 
-	toHeader, err := h.syncer.Client.GetHeader(toHeaderBlockRoot)
-	if err != nil {
-		return err
-	}
+	return nil
+}
 
-	fromSlot := fromHeader.Slot
-	toSlot := toHeader.Slot
-	totalSlots := toSlot - fromSlot
+func (h *Header) SyncHeaders(ctx context.Context, fromSlot, toSlot uint64) error {
 
 	log.WithFields(log.Fields{
-		"fromHeader": fromHeaderBlockRoot,
 		"fromSlot":   fromSlot,
 		"fromEpoch":  h.syncer.ComputeEpochAtSlot(fromSlot),
-		"toHeader":   toHeaderBlockRoot,
 		"toSlot":     toSlot,
 		"toEpoch":    h.syncer.ComputeEpochAtSlot(toSlot),
-		"totalSlots": totalSlots,
+		"totalSlots": toSlot - fromSlot,
 	}).Info("starting to back-fill headers")
 
-	headersToSync := []scale.HeaderUpdate{}
+	var headersToSync []scale.HeaderUpdate
 
-	currentSlot := fromSlot + 1 // start syncing at next block after last synced block
-
-	checkpoint, err := h.getClosestCheckpoint(currentSlot)
-	if err != nil {
-		return fmt.Errorf("get closest checkpoint: %w", err)
-	}
-
-	headerUpdate, err := h.syncer.GetNextHeaderUpdateBySlotWithAncestryProof(currentSlot, checkpoint)
+	// start syncing at next block after last synced block
+	currentSlot := fromSlot
+	headerUpdate, err := h.getNextHeaderUpdateBySlotWithAncestryProof(currentSlot)
 	if err != nil {
 		return fmt.Errorf("get next header update by slot with ancestry proof: %w", err)
 	}
 
 	for currentSlot <= toSlot {
-		epoch := h.syncer.ComputeEpochAtSlot(currentSlot)
-
-		currentSyncPeriod := h.syncer.ComputeSyncPeriodAtSlot(currentSlot)
-
-		if currentSyncPeriod > h.cache.LastSyncedSyncCommitteePeriod {
-			err = h.syncLaggingSyncCommitteePeriods(ctx, h.cache.LastSyncedSyncCommitteePeriod, currentSlot)
-			if err != nil {
-				return fmt.Errorf("sync lagging sync committee periods: %w", err)
-			}
-		}
-
-		// start of new epoch, sync headers of last epoch
-		if h.syncer.IsStartOfEpoch(currentSlot) {
-			log.WithFields(log.Fields{
-				"epoch": epoch - 1,
-			}).Debug("syncing header in epoch")
-			for _, header := range headersToSync {
-				err := h.SyncHeader(ctx, header, toSlot-uint64(header.Payload.Header.Slot))
-				if err != nil {
-					return fmt.Errorf("sync execution header: %w", err)
-				}
-			}
```

Comment:
no need to sync header for each epoch, as the max latency is no more than 1 sync_period now.